### PR TITLE
Flexible Digestion Damage

### PR DESCRIPTION
**Pull request by SpringSkipper**
Lets you pick any amount of damage up to a combined total of 36 for belly digestion damage, rather than a max of 6/6/12/6/6

Damage above the normal limit is already something you can ahelp to be given, and keeping the same overall damage limit means that this isn't particularly more exploitable by bad actors or problem players. The only real difference would be better availability of bellies with bone-crunchy noises. Which... if we don't trust our players even that much then what's the point?

This also adds a digestion_max variable to voregans that controls the maximum total values allowed, and it can be changed in the view variables menu by admins. This value saves alongside a belly (but cannot be exported/imported), and persists between rounds if changed.
## Changelog
:cl:
qol: digestion damage types can be set higher; the max total damage is still the same.
/:cl:

### CODE REVIEW

**Comment by Aroliacue** [comment]
This is my favourite PR

File: code/modules/vore/eating/vorepanel_vr.dm [hunk 1]
```diff
@@ -3263,40 +3265,85 @@ var/global/list/belly_colorable_only_fullscreens = list("a_synth_flesh_mono",
 			host.vore_selected.nutrition_percent = new_new_nutrition
 			. = TRUE
 		if("b_burn_dmg")
-			var/new_damage = tgui_input_number(user, "Choose the amount of burn damage prey will take per tick. Ranges from 0 to 6.", "Set Belly Burn Damage.", host.vore_selected.digest_burn, 6, 0, round_value=FALSE)
+			// var/new_damage = tgui_input_number(user, "Choose the amount of burn damage prey will take per tick. Ranges from 0 to 6.", "Set Belly Burn Damage.", host.vore_selected.digest_burn, 6, 0, round_value=FALSE)
+			// CHOMP Edit
+			var/old_damage = host.vore_selected.digest_burn
+			var/max_total_damage = host.vore_selected.digest_max
+			var/current_total_damage = host.vore_selected.get_total_digestion_damage()
+			var/new_damage = tgui_input_number(user, "Choose the amount of burn damage prey will take per tick. Max of 36 across all damage types.", "Set Belly Burn Damage.", old_damage, max_total_damage, 0, round_value=FALSE)
```
Comment (Kashargul) [inline code/modules/vore/eating/vorepanel_vr.dm:3273]: No, those shouldn't use the max, those should be calculated dynamically in this case.

File: code/modules/vore/eating/vorepanel_vr.dm [hunk 2]
```diff
@@ -3263,40 +3265,85 @@ var/global/list/belly_colorable_only_fullscreens = list("a_synth_flesh_mono",
...
 			host.vore_selected.items_preserved.Cut() //CHOMPAdd
 			. = TRUE
 		if("b_brute_dmg")
-			var/new_damage = tgui_input_number(user, "Choose the amount of brute damage prey will take per tick. Ranges from 0 to 6", "Set Belly Brute Damage.", host.vore_selected.digest_brute, 6, 0, round_value=FALSE)
+			// var/new_damage = tgui_input_number(user, "Choose the amount of brute damage prey will take per tick. Ranges from 0 to 6", "Set Belly Brute Damage.", host.vore_selected.digest_brute, 6, 0, round_value=FALSE)
+			// CHOMP Edit
+			var/old_damage = host.vore_selected.digest_brute
+			var/max_total_damage = host.vore_selected.digest_max
+			var/current_total_damage = host.vore_selected.get_total_digestion_damage()
+			var/new_damage = tgui_input_number(user, "Choose the amount of brute damage prey will take per tick. Max of 36 across all damage types.", "Set Belly Brute Damage.", old_damage, max_total_damage, 0, round_value=FALSE)
```
Comment (Kashargul) [inline code/modules/vore/eating/vorepanel_vr.dm:3290]: As before..

File: code/modules/vore/eating/vorepanel_vr.dm [hunk 3]
```diff
@@ -3263,40 +3265,85 @@ var/global/list/belly_colorable_only_fullscreens = list("a_synth_flesh_mono",
...
 			host.vore_selected.items_preserved.Cut() //CHOMPAdd
 			. = TRUE
 		if("b_oxy_dmg")
-			var/new_damage = tgui_input_number(user, "Choose the amount of suffocation damage prey will take per tick. Ranges from 0 to 12.", "Set Belly Suffocation Damage.", host.vore_selected.digest_oxy, 12, 0, round_value=FALSE)
+			// var/new_damage = tgui_input_number(user, "Choose the amount of suffocation damage prey will take per tick. Ranges from 0 to 12.", "Set Belly Suffocation Damage.", host.vore_selected.digest_oxy, 12, 0, round_value=FALSE)
+			// CHOMP Edit
+			var/old_damage = host.vore_selected.digest_oxy
+			var/max_total_damage = host.vore_selected.digest_max
+			var/current_total_damage = host.vore_selected.get_total_digestion_damage()
+			var/new_damage = tgui_input_number(user, "Choose the amount of suffocation damage prey will take per tick. Max of 36 across all damage types.", "Set Belly Suffocation Damage.", old_damage, max_total_damage, 0, round_value=FALSE)
```
Comment (Kashargul) [inline code/modules/vore/eating/vorepanel_vr.dm:3307]: As before..

File: code/modules/vore/eating/vorepanel_vr.dm [hunk 4]
```diff
@@ -3263,40 +3265,85 @@ var/global/list/belly_colorable_only_fullscreens = list("a_synth_flesh_mono",
...
 			host.vore_selected.digest_oxy = new_new_damage
 			. = TRUE
 		if("b_tox_dmg")
-			var/new_damage = tgui_input_number(user, "Choose the amount of toxins damage prey will take per tick. Ranges from 0 to 6", "Set Belly Toxins Damage.", host.vore_selected.digest_tox, 6, 0, round_value=FALSE)
+			// var/new_damage = tgui_input_number(user, "Choose the amount of toxins damage prey will take per tick. Ranges from 0 to 6", "Set Belly Toxins Damage.", host.vore_selected.digest_tox, 6, 0, round_value=FALSE)
+			// CHOMP Edit
+			var/old_damage = host.vore_selected.digest_tox
+			var/max_total_damage = host.vore_selected.digest_max
+			var/current_total_damage = host.vore_selected.get_total_digestion_damage()
+			var/new_damage = tgui_input_number(user, "Choose the amount of toxins damage prey will take per tick. Max of 36 across all damage types.", "Set Belly Toxins Damage.", old_damage, max_total_damage, 0, round_value=FALSE)
```
Comment (Kashargul) [inline code/modules/vore/eating/vorepanel_vr.dm:3323]: As before..

File: code/modules/vore/eating/vorepanel_vr.dm [hunk 5]
```diff
@@ -3263,40 +3265,85 @@ var/global/list/belly_colorable_only_fullscreens = list("a_synth_flesh_mono",
...
 			host.vore_selected.digest_tox = new_new_damage
 			. = TRUE
 		if("b_clone_dmg")
-			var/new_damage = tgui_input_number(user, "Choose the amount of brute DNA damage (clone) prey will take per tick. Ranges from 0 to 6", "Set Belly Clone Damage.", host.vore_selected.digest_clone, 6, 0, round_value=FALSE)
+			// var/new_damage = tgui_input_number(user, "Choose the amount of brute DNA damage (clone) prey will take per tick. Ranges from 0 to 6", "Set Belly Clone Damage.", host.vore_selected.digest_clone, 6, 0, round_value=FALSE)
+			// CHOMP Edit
+			var/old_damage = host.vore_selected.digest_clone
+			var/max_total_damage = host.vore_selected.digest_max
+			var/current_total_damage = host.vore_selected.get_total_digestion_damage()
+			var/new_damage = tgui_input_number(user, "Choose the amount of brute DNA damage (clone) prey will take per tick. Max of 36 across all damage types.", "Set Belly Clone Damage.", old_damage, max_total_damage, 0, round_value=FALSE)
```
Comment (Kashargul) [inline code/modules/vore/eating/vorepanel_vr.dm:3339]: As before..

**Review by Kashargul — CHANGES_REQUESTED** [review]

File: code/modules/vore/eating/vorepanel_vr.dm (hunk 1 above)
Comment (SpringSkipper) [reply code/modules/vore/eating/vorepanel_vr.dm]: Redid those entirely to be a lot cleaner.

**Comment by Kashargul** [comment]
Do NEVER remove the sanity checks on those, the CLAMP must remain. Due to how TGUI Number input works, you will break it if default is > maximum.

**Review by Kashargul — APPROVED** [review]

**Pull request merged**